packages/truffle-config/index.js

```diff
@@ -390,4 +390,13 @@ Config.getUserConfig = function() {
   return new Configstore("truffle", {}, { globalConfigPath: true });
 };
 
+Config.getCacheDirectory = () => {
```

Comment:
I think using the word `cache` here is a bit ambiguous. Maybe we could call this the "user config directory"? Or we could make this less general and just make the method `getCompilersCacheDirectory`. Thoughts?

Reply:
It seems to me that user configuration might be a more specific use for the cache directory. In other words, Truffle has a directory that it uses to cache things. One of those things happens to be the user configuration data. And I would be totally fine with making it `Config.getCompilersCacheDirectory`. That way the loading strategies don't need to do any legwork and `Config` can worry about that stuff.

Reply:
My thinking here is predicated on my sense that the word "cache" implies something that can be deleted without losing information. E.g., the compilers are cached because they can be redownloaded. I would say that the user config isn't cached because if it's deleted, it can't be recreated automatically.

But I may be defining "cache" too specifically in my head, so I'd love a second opinion.

Reply:
Oh I see what you mean. Maybe it should be something more like `truffleDataDirectory`. Maybe inside there can be a folder named something like `cachedCompilers`...how would that be?

Reply:
both those sound good!

Reply:
Actually I guess `Config` shouldn't necessarily know about the compiler cacheing path...the loading strategy should handle that I think.

Reply:
I think Config should know about the "truffle data directory", but the supplier manages the compiler cache itself